src/rgw/rgw_http_errors.h

```diff
@@ -10,65 +10,66 @@ struct rgw_http_errors {
-    { ERR_INVALID_TENANT_NAME, 400, "InvalidTenantName" },
-    { ENOTEMPTY, 409, "BucketNotEmpty" },
-    { ERR_PRECONDITION_FAILED, 412, "PreconditionFailed" },
-    { ERANGE, 416, "InvalidRange" },
-    { ERR_UNPROCESSABLE_ENTITY, 422, "UnprocessableEntity" },
-    { ERR_LOCKED, 423, "Locked" },
-    { ERR_INTERNAL_ERROR, 500, "InternalError" },
-    { ERR_NOT_IMPLEMENTED, 501, "NotImplemented" },
-    { ERR_SERVICE_UNAVAILABLE, 503, "ServiceUnavailable"}
+    { 0, 200, "", "" },
+    { STATUS_CREATED, 201, "Created", "Created" },
+    { STATUS_ACCEPTED, 202, "Accepted", "Accepted" },
+    { STATUS_NO_CONTENT, 204, "NoContent", "No content" },
+    { STATUS_PARTIAL_CONTENT, 206, "", ""  },
+    { ERR_PERMANENT_REDIRECT, 301, "PermanentRedirect", "The bucket you are attempting to access must be addressed using the specified endpoint. Send all future requests to this endpoint." },
+    { ERR_WEBSITE_REDIRECT, 301, "WebsiteRedirect", "Website redirect" },
+    { STATUS_REDIRECT, 303, "", "Status redirect" },
+    { ERR_NOT_MODIFIED, 304, "NotModified", "Not modified" },
+    { EINVAL, 400, "InvalidArgument", "Invalied Argument" },
+    { ERR_INVALID_REQUEST, 400, "InvalidRequest", "Please use AWS4-HMAC-SHA256; SOAP requests must be made over an HTTPS connection." },
+    { ERR_INVALID_DIGEST, 400, "InvalidDigest", "The Content-MD5 you specified is not valid." },
+    { ERR_BAD_DIGEST, 400, "BadDigest", "The Content-MD5 you specified did not match what we received." },
+    { ERR_INVALID_BUCKET_NAME, 400, "InvalidBucketName", "The specified bucket is not valid." },
+    { ERR_INVALID_OBJECT_NAME, 400, "InvalidObjectName", "The specified object is not valid." },
+    { ERR_UNRESOLVABLE_EMAIL, 400, "UnresolvableGrantByEmailAddress", "The email address you provided does not match any account on record." },
+    { ERR_INVALID_PART, 400, "InvalidPart", "One or more of the specified parts could not be found. The part might not have been uploaded, or the specified entity tag might not have matched the part's entity tag." },
+    { ERR_INVALID_PART_ORDER, 400, "InvalidPartOrder", "The list of parts was not in ascending order.Parts list must specified in order by part number." },
+    { ERR_REQUEST_TIMEOUT, 400, "RequestTimeout", "Your socket connection to the server was not read from or written to within the timeout period." },
+    { ERR_TOO_LARGE, 400, "EntityTooLarge", "Your proposed upload exceeds the maximum allowed object size." },
+    { ERR_TOO_SMALL, 400, "EntityTooSmall", "Your proposed upload is smaller than the minimum allowed object size." },
+    { ERR_TOO_MANY_BUCKETS, 400, "TooManyBuckets", "You have attempted to create more buckets than allowed." },
+    { ERR_MALFORMED_XML, 400, "MalformedXML", "This happens when the user sends malformed xml (xml that doesn't conform to the published xsd) for the configuration. The error message is, \"The XML you provided was not well-formed or did not validate against our published schema.\"" },
```

Comment:
This seems overly specific. We do not seem to actually return this error at present. Suggest something more generic like: `"The XML you provided was not well-formed or did not validate against our published schema."`